trajectory_processing/src/iterative_time_parameterization.cpp

```diff
@@ -111,13 +112,18 @@ void IterativeParabolicTimeParameterization::applyVelocityConstraints(robot_traj
   {
     const robot_state::RobotStatePtr &curr_waypoint = rob_trajectory.getWayPointPtr(i);
     const robot_state::RobotStatePtr &next_waypoint = rob_trajectory.getWayPointPtr(i+1);
-    
+
+    double velocity_scaling_factor = 1.0;
+
+    if (max_velocity_scaling_factor > 0.0 && max_velocity_scaling_factor <= 1.0)
+      velocity_scaling_factor = max_velocity_scaling_factor;
```

Comment:
This user error checking silently fails, I think it should produce a warning using logWarn. Or maybe we could remove this all together and just use an assert to improve performance.